analyses/chromosomal-instability/02b-plot-chr-instability-by-histology.Rmd

```diff
@@ -0,0 +1,195 @@
+
+```{r}
+# Let's check if this is numeric
+if (!(is.numeric(params$min_samples))) {
+  warning("The parameter min_samples should be numeric. Trying to coerce to numeric.")
+  params$min_samples <- as.numeric(params$min_samples)
+} 
+```
+
+### Directories and Files
+
+```{r}
+# Path to data directory
+data_dir <- file.path("..", "..", "data")
+
+# Path to output directory
+plots_dir <- "plots"
+
+# Path to tumor type plots output directory
+hist_plots_dir <- file.path(plots_dir, "tumor-type")
+
+# Create the hist_plots_dir  if it does not exist
+if (!dir.exists(hist_plots_dir)) {
+  dir.create(hist_plots_dir, recursive = TRUE)
+}
+```
+
+Here's all the input files we will need: 
+
+```{r}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+hist_binned_breaks_file <- file.path("breakpoint-data", "histology_breakpoint_binned_counts.RDS")
+total_densities_file <- file.path("breakpoint-data", "intersection_of_breaks_densities.tsv")
+```
+
+Output plot: 
+
+```{r}
+cdf_plot_file <- file.path(plots_dir, "breaks_cdf_plot.png")
+```
+
+### Read in data 
+
+Set up metadata
+
+```{r}
+# Read in the metadata
+metadata <- readr::read_tsv(metadata_file)
+```
+
+Load in the localized by histology breakpoint data. 
+
+```{r}
+hist_break_local <- readr::read_rds(hist_binned_breaks_file)
+```
+
+Read in the total intersection breaks densities TSV that was made by the 
+`00-setup-breakpoint-data.R` script. 
+
+```{r}
+total_break_densities_df <- readr::read_tsv(
+  total_densities_file
+) %>%
+  # Add on the short histology column from the metadata for plotting purposes
+  dplyr::inner_join(metadata %>% dplyr::select(Kids_First_Biospecimen_ID, short_histology),
+    by = c("samples" = "Kids_First_Biospecimen_ID")
+  )
+```
+
+## Plot CDF plot by histology
+
+```{r}
+breaks_cdf <- total_break_densities_df  %>%
+  # Only plot histologies groups with more than `min_samples` number of samples
+  dplyr::group_by(short_histology, add = TRUE) %>%
+  dplyr::filter(dplyr::n() > params$min_samples) %>%
+  # Calculate histology group mean
+  dplyr::mutate(
+    hist_mean = mean(breaks_density),
+    hist_rank = rank(breaks_density) / dplyr::n()
+  ) %>%
+  dplyr::ungroup() %>%
+  # Now we will plot these as cummulative distribution plots
+  ggplot2::ggplot(ggplot2::aes(
+    x = hist_rank, 
+    y = breaks_density,
+    color = short_histology
+  )) +
+  ggplot2::geom_point() +
+  # Add summary line for mean
+  ggplot2::geom_hline(ggplot2::aes(yintercept = hist_mean)) +
+  # Separate by histology
+  ggplot2::facet_wrap(~short_histology, nrow = 1) +
+  ggplot2::theme_classic() +
+  ggplot2::xlab("") +
+  ggplot2::ylab("Chromosomal Breaks per Mb") +
+  # Transform to log10 make non-log y-axis labels
+  ggplot2::scale_y_continuous(trans = "log10", breaks = c(0, 1, 10, 30)) +
```

Comment:
The breaks aren't showing up, presumably because "breaks_density" is tiny. Lets just plot total breaks/sample? label as Breaks per genome?